source/developer/personal-access-tokens.rst

```diff
@@ -0,0 +1,92 @@
+.. _personal_access_tokens:
+
+Personal Access Tokens
+=======================
+
+Personal access tokens function similar to session tokens and can be used by integrations to `authenticate against the REST API <https://about.mattermost.com/default-api-authentication>`_. It is the most commonly used type of token for integrations.
+
+.. toctree::
+	:maxdepth: 2
+
+Creating a Personal Access Token
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+1 - Enable personal access tokens in **System Console > Integrations > Custom Integrations**.
+
+2 - Identify the account you want to create a personal access token with. You may optionally create a new user account for your integration, such as for a bot account. By default, only System Admins have permissions to create a personal access token. 
+
+3 - To create an access token with a non-admin account, you must first give it the appropriate permissions. Go to **System Console > Users**, search for the user account, and select **Manage Roles** from the dropdown.
+
+	.. image:: ../../source/images/access_token_manage_roles.png
+
+4 - Select **Allow this account to generate user access tokens.**
+
+	.. image:: ../../source/images/access_tokens_additional_roles.png
+  
+  // XXX This image needs UI tweaks from Asaad
+
+You may optionally allow the account to post to any channel in your Mattermost server, including direct messages by choosing the **post:all** role. **post:channels** role allows the account to post to any public channel in the Mattermost server.
+
+Then hit **Save**.
+
+5 - Sign in to the user account to create a personal access token.
+
+6 - Go to **Account Settings > Security > Personal Access Tokens** and click **Create New Token**.
+
+  // XXX Take a screenshot
+
+7 - Enter a description for the token, so you remember what it's used for. Then hit **Save**.
+
+.. note::
+  If you create a personal access token for a System Admin account, be careful who you share it with. The token enables a user to have full access to the account, including System Admin privileges.
```

Comment:
Maybe `be careful` to `be extra careful`, no matter who you are you should be careful with your access tokens